pkg/resource/any_test.go

```diff
@@ -30,6 +31,26 @@ func TestNewAnyFromProto(t *testing.T) {
 	r, err := resource.NewAnyFromProto(&protoMd{}, &protoSpec{})
 	assert.NoError(t, err)
 
-	assert.Equal(t, map[string]interface{}{"something": []interface{}{"a", "b", "c"}, "value": "xyz"}, r.Spec())
+	assert.Equal(t, map[string]interface{}{"something": []interface{}{"a", "b", "c"}, "value": "xyz"}, r.Value())
 	assert.Equal(t, "aaa", r.Metadata().ID())
+
+	enc, err := resource.MarshalYAML(r)
+	assert.NoError(t, err)
+
+	out, err := yaml.Marshal(enc)
+	assert.NoError(t, err)
+
+	assert.Equal(t, `metadata:
+    namespace: default
+    type: type
+    id: aaa
+    version: 1
+    phase: running
+    finalizers:
+      - resource1
+      - resource2
+spec:
+    value: xyz
+    something: [a, b, c]
```

Comment:
this is not the default formatting, so this verifies that original flow-style is preserved here